src/components/MaIcon/MaIcon.vue

```diff
@@ -109,7 +90,28 @@ export default {
     },
 
     iconName() {
-      return this.title ? this.title : `${this.icon}-icon`
+      return this.title ? this.title : `${this.icon}Icon`
+    },
+
+    sizingProperties() {
+      if (this.size === 'small') {
+        return {
+          width: '16',
+          height: '16',
+          viewBox: '0 0 24 24',
+        }
+      } else if (this.size === 'medium') {
+        return {
+          width: '24',
+          height: '24',
+          viewBox: '0 0 24 24',
+        }
+      }
+      return {
+        width: '32',
+        height: '32',
+        viewBox: '0 0 24 24',
+      }
```

Comment:
Following the discussion we had over meets, the following code is equivalent and it probably looks a bit better (this is totally subjective, obviously 😄):

```suggestion
      const sizingProperties = {
        small: {
          width: '16',
          height: '16',
          viewBox: '0 0 24 24',
        },
        medium: {
          width: '24',
          height: '24',
          viewBox: '0 0 24 24',
        },
        large: {
          width: '32',
          height: '32',
          viewBox: '0 0 24 24',
        },
      }

      return sizingProperties[this.size]
```

the result is exactly the same (hopefully 😂) but we avoid some repetition and an if/else structure. Instead, we create an object with 3 keys – small, medium, large – with the values for each. Then, depending on the value of `this.size`, we return the values from the appropriate key.

Yet another improvement is to define the "list" of values in the `data` section of the component, and then the `computed` would only be `return this.properties[this.size]`.

In any case, please remember this is simply a refactor and that the behavior is perfectly fine as is 👌  great job

Reply:
Good point! Should this be properly allocated with the rest of our ds tokens? Looks like these design rules could live with the rest of them (even though right now we only use them in the icon, we could end up having `ma-img`?)

Reply:
premature abstraction? 😬  they are only used here and they don't change responsively, so I don't really see the benefit of porting them to an external file (not yet, at least!)

Reply:
Well... if we change the used framework, will these values change? 😬  I wasn't saying we should do these now, but it would be nice to have all the rules that define our Design System apart and isolated from the rest of the framework code.

Reply:
fair points!